src/Common/TimerDescriptor.cpp

```diff
@@ -19,15 +20,13 @@ namespace ErrorCodes
 
 TimerDescriptor::TimerDescriptor(int clockid, int flags)
 {
-    timer_fd = timerfd_create(clockid, flags);
+    timer_fd = timerfd_create(clockid, flags | TFD_NONBLOCK);
     if (timer_fd == -1)
-        throw Exception(ErrorCodes::CANNOT_CREATE_TIMER, "Cannot create timer_fd descriptor");
```

Comment:
It's strange that here it was Exception, but next two lines, it was ErrnoException.